app/controllers/idv/sessions_controller.rb

```diff
@@ -4,125 +4,41 @@ class SessionsController < ApplicationController
 
     before_action :confirm_two_factor_authenticated
 
-    def index
+    helper_method :idv_profile_form
+
+    def new
       @using_mock_vendor = pick_a_vendor == :mock
-      @profile = idv_profile_form
     end
 
     def create
       idv_params.merge!(profile_params)
-      @profile = idv_profile_form
       submit_profile
     end
 
     def dupe
     end
 
-    def finance
-      prep_step
-    end
-
-    def phone
-      prep_step
-    end
-
-    def review
-      prep_step
-    end
-
-    def update_finance
-      prep_step
-      idv_params.merge!(financial_params.delete_if { |_key, value| value.blank? })
-      redirect_to idv_phone_url
-    end
-
-    def update_review
-      self.idv_applicant = applicant_from_params
-      resolution = start_idv_session
-      if resolution.success
-        init_questions_and_profile(resolution)
-        redirect_on_success
-      else
-        flash[:error] = I18n.t('idv.titles.fail')
-        redirect_to idv_sessions_url
-      end
-    end
-
     private
 
     def submit_profile
-      if @profile.submit(profile_params)
-        redirect_to idv_sessions_finance_url
-      elsif @profile.errors.include?(:ssn)
+      if idv_profile_form.submit(profile_params)
+        redirect_to idv_finance_url
+      elsif idv_profile_form.errors.include?(:ssn)
```

Comment:
There are multiple reasons the SSN could be invalid, but we only want to redirect if the error is related to the SSN not being unique. A blank SSN will trigger this as well. You can try it by submitting the form without entering anything (with JS disabled).